internal/handlers/pg/pgdb/query_documents_test.go

```diff
@@ -0,0 +1,164 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Use _test package to avoid import cycle with testutil.
+package pgdb_test
+
+import (
+	"testing"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+	"golang.org/x/net/context"
+
+	"github.com/FerretDB/FerretDB/internal/handlers/pg/pgdb"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+	"github.com/FerretDB/FerretDB/internal/util/testutil"
+)
+
+func TestQueryDocuments(t *testing.T) {
+	t.Parallel()
+
+	ctx := testutil.Ctx(t)
+
+	pool := getPool(ctx, t, nil, zaptest.NewLogger(t))
+	dbName := testutil.DatabaseName(t)
+	collectionName := testutil.CollectionName(t)
+
+	t.Cleanup(func() {
+		pool.DropDatabase(ctx, dbName)
+	})
+
+	pool.DropDatabase(ctx, dbName)
+	require.NoError(t, pgdb.CreateDatabase(ctx, pool, dbName))
+
+	cases := []struct {
+		name             string
+		collection       string
+		documents        []*types.Document
+		docsPerIteration []int // how many documents should be fetched per each iteration
+		// len(docsPerIteration) is the amount of fetch iterations.
+	}{
+		{
+			name:             "empty",
+			collection:       collectionName,
+			documents:        []*types.Document{},
+			docsPerIteration: []int{},
+		},
+		{
+			name:             "one",
+			collection:       collectionName + "_one",
+			documents:        []*types.Document{must.NotFail(types.NewDocument("id", "1"))},
+			docsPerIteration: []int{1},
+		},
+		{
+			name:       "two",
+			collection: collectionName + "_two",
+			documents: []*types.Document{
+				must.NotFail(types.NewDocument("id", "1")),
+				must.NotFail(types.NewDocument("id", "2")),
+			},
+			docsPerIteration: []int{2},
+		},
+		{
+			name:       "three",
+			collection: collectionName + "_three",
+			documents: []*types.Document{
+				must.NotFail(types.NewDocument("id", "1")),
+				must.NotFail(types.NewDocument("id", "2")),
+				must.NotFail(types.NewDocument("id", "3")),
+			},
+			docsPerIteration: []int{2, 1},
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			err := pool.InTransaction(ctx, func(tx pgx.Tx) error {
+				for _, doc := range tc.documents {
+					require.NoError(t, pgdb.InsertDocument(ctx, tx, dbName, tc.collection, doc))
+				}
+
+				fetchedChan, err := pool.QueryDocuments(ctx, tx, dbName, tc.collection, "")
+				require.NoError(t, err)
+
+				iter := 0
+				for {
+					fetched, ok := <-fetchedChan
+					if !ok {
+						break
+					}
+
+					assert.NoError(t, fetched.Err)
+					assert.Equal(t, tc.docsPerIteration[iter], len(fetched.Docs))
+					iter++
+				}
+				assert.Equal(t, len(tc.docsPerIteration), iter)
+
+				return nil
+			})
+
+			require.NoError(t, err)
+		})
+	}
+
+	// Special case: cancel context before reading from channel.
```

Comment:
Let's remove unused code and extract it to the TODO.

Reply:
Yeah, that test actually works, I just forgot to uncomment it after some troubleshooting 🙈
Fixed!